api/handler/object_list.go

```diff
@@ -0,0 +1,278 @@
+package handler
+
+import (
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/nspcc-dev/neofs-s3-gw/api"
+	"github.com/nspcc-dev/neofs-s3-gw/api/layer"
+)
+
+// ListObjectsV1Handler handles objects listing requests for API version 1.
+func (h *handler) ListObjectsV1Handler(w http.ResponseWriter, r *http.Request) {
+	params, err := parseListObjectsArgsV1(r)
+	if err != nil {
+		h.registerAndSendError(w, r, err, "failed to parse arguments")
+		return
+	}
+
+	list, err := h.obj.ListObjectsV1(r.Context(), params)
+	if err != nil {
+		h.registerAndSendError(w, r, err, "something went wrong")
+		return
+	}
+
+	err = api.EncodeToResponse(w, encodeV1(params, list))
+	if err != nil {
+		h.registerAndSendError(w, r, err, "something went wrong")
+	}
+}
+
+func encodeV1(p *layer.ListObjectsParamsV1, list *layer.ListObjectsInfoV1) *ListObjectsV1Response {
+	res := &ListObjectsV1Response{
+		Name:         p.Bucket,
+		EncodingType: p.Encode,
+		Marker:       p.Marker,
+		Prefix:       p.Prefix,
+		MaxKeys:      p.MaxKeys,
+		Delimiter:    p.Delimiter,
+
+		IsTruncated: list.IsTruncated,
+		NextMarker:  list.NextMarker,
+	}
+
+	// fill common prefixes
+	for i := range list.Prefixes {
+		res.CommonPrefixes = append(res.CommonPrefixes, CommonPrefix{
+			Prefix: s3PathEncode(list.Prefixes[i], p.Encode),
+		})
+	}
+
+	// fill contents
+	for _, obj := range list.Objects {
```

Comment:
Can we separate this into a separate function and use in `encodeV2`?

Reply:
Moved filling prefixes and contents into separate functions.